client/iface/bind/control_android.go

```diff
@@ -0,0 +1,12 @@
+package bind
+
+import (
+	wireguard "golang.zx2c4.com/wireguard/conn"
+
+	nbnet "github.com/netbirdio/netbird/util/net"
+)
+
+func init() {
+	// ControlFns is not thread safe and should only be modified during init.
+	*wireguard.ControlFns = append(*wireguard.ControlFns, nbnet.ControlProtectSocket)
```

Comment:
Should we check for nil here first? Just in case lib code changes

Reply:
hmm, hard question what is the best error handling here. If it is nil what can we do? We are in the init() function. Throw a fatal error? Actually we do it right now just without error check.

Reply:
I don't have a good idea either. Let's create a ticket to handle this later